php/Block.php

```diff
@@ -63,66 +63,179 @@ public function register_block() {
+			$posts_list_query->found_posts   = $posts_list_query->found_posts - 1;
+		}
+
+		// Block render callbacks expect a string, so begin capturing any direct
+		// output.
+		ob_start();
 		?>
-        <div class="<?php echo $class_name; ?>">
-			<h2>Post Counts</h2>
-			<ul>
-			<?php
-			foreach ( $post_types as $post_type_slug ) :
-                $post_type_object = get_post_type_object( $post_type_slug  );
-                $post_count = count(
-                    get_posts(
-						[
-							'post_type' => $post_type_slug,
-							'posts_per_page' => -1,
-						]
-					)
-                );
+		<div class="<?php echo esc_attr( $attributes['className'] ?? '' ); ?>">
+			<h2><?php esc_html_e( 'Post Counts', 'site-counts' ); ?></h2>
 
+			<?php
+			/**
+			 * Display a list of public post types and how many posts each has.
+			 */
+			if ( ! empty( $public_post_type_counts ) ) :
 				?>
-				<li><?php echo 'There are ' . $post_count . ' ' .
-					  $post_type_object->labels->name . '.'; ?></li>
-			<?php endforeach;	?>
-			</ul><p><?php echo 'The current post ID is ' . $_GET['post_id'] . '.'; ?></p>
+					<ul>
+						<?php foreach ( $public_post_type_counts as $values ) : ?>
+							<li>
+								<?php
+								echo esc_html(
+									sprintf(
+										__( 'There are %1$d %2$s.', 'site-counts' ),
+										$values['count'] ?? 0,
+										$values['label'] ?? ''
+									)
+								);
+								?>
+							</li>
+						<?php endforeach; ?>
+					</ul>
+				<?php
+			endif;
 
-			<?php
-			$query = new WP_Query(  array(
-				'post_type' => ['post', 'page'],
-				'post_status' => 'any',
-				'date_query' => array(
-					array(
-						'hour'      => 9,
-						'compare'   => '>=',
-					),
-					array(
-						'hour' => 17,
-						'compare'=> '<=',
-					),
-				),
-                'tag'  => 'foo',
-                'category_name'  => 'baz',
-				  'post__not_in' => [ get_the_ID() ],
-			));
-
-			if ( $query->have_posts() ) :
+			/**
+			 * Display the current post ID.
+			 */
+			?>
+			<p>
+				<?php
+				echo esc_html(
+					sprintf(
+						__( 'The current post ID is %1$d.', 'site-counts' ),
+						absint( $current_post_id )
+					)
+				);
 				?>
-				 <h2>5 posts with the tag of foo and the category of baz</h2>
-                <ul>
-                <?php
+			</p>
+			<?php
 
-                 foreach ( array_slice( $query->posts, 0, 5 ) as $post ) :
-                    ?><li><?php echo $post->post_title ?></li><?php
-				endforeach;
-			endif;
-		 	?>
-			</ul>
+			/**
+			 * Display post titles for up to 5 posts & pages published between 9:00 and 17:00,
+			 * with the tag (slug) `foo`, and category (name) `baz`.
+			 */
+			if ( $posts_list_query->have_posts() ) :
+				?>
+				<h2><?php
+					echo esc_html(
+						sprintf(
+							__( '%1$d posts with the tag of foo and the category of baz', 'site-counts' ),
+							absint( $posts_list_query->found_posts )
+						)
+					);
+				?></h2>
```

Comment:
The php block could go onto new lines and be tabbed to align, but I'm also aware of the odd whitespace that comes with it. Personally preference really, but I'd follow project/company code standards and look for some feedback in the code review to gage other's opinons.